backend/molgenis-emx2-rdf/src/main/java/org/molgenis/emx2/rdf/ColumnTypeRdfMapper.java

```diff
@@ -274,10 +293,39 @@ boolean isEmpty(Row row, Column column) {
       }
     },
     ONTOLOGY(CoreDatatype.XSD.ANYURI) {
-      // TODO: Implement Ontology behavior where it also returns ontologyTermURI as Value.
       @Override
       Set<Value> retrieveValues(String baseURL, Row row, Column column) {
-        return RdfColumnType.REFERENCE.retrieveValues(baseURL, row, column);
+        final TableMetadata target = column.getRefTable();
+        final String rootTableName =
+            UrlEscapers.urlPathSegmentEscaper().escape(target.getRootTable().getIdentifier());
+        final Namespace ns = getSchemaNamespace(baseURL, target.getRootTable().getSchema());
+
+        String[] names =
+            (column.isArray()
+                ? row.getStringArray(column.getName())
+                : new String[] {row.getString(column.getName())});
+        Filter[] filters =
+            Arrays.stream(names).map(i -> f("name", EQUALS, i)).toArray(Filter[]::new);
+
+        List<Row> refRows =
+            column
+                .getRefTable()
+                .getTable()
+                .query()
+                .select(s("name"), s("ontologyTermURI"))
```

Comment:
oei, dat is erg duur om dat per stuk te doen. Komt dit vaak voor? Want dan zou je liever de originele query expanden?

Reply:
New approach implemented where the `name` & `ontologyTermURI` of referenced ontology tables are stored. This is done once per API-call.